### ISSUE TITLE
capture OS and Node.js specific diagnostic information when errors are thrown

### ISSUE THREAD

**Issue by tucker01** [zowe/imperative#76]
-  [x]  Determine diagnostic details to collect
          Complete
 -  [x]  Code collection of diagnostic details 
          Complete 

After removing "node report", we should attempt to capture environmental information using `process` and `require("os")` when an error is thrown. 

Having this information handy in logs avoids having to ask for it when issues occur.

**Comment by dkelosky** [comment]
Also the stack!

**Comment by AHumanFromCA** [comment]
Don't forget the exact command executed in the CLI and environment variables. node_report did this:

```
Command line: node  C:\Users\XXXXXXX\Documents\GitHub\zowe-cli\lib\main.js bsp fail eh
```

**Comment by ChrisBoehmCA** [comment]
Hopefully addressed by #117 where log messages were added
